metafields.go

```diff
@@ -30,9 +30,10 @@ func (m Metafields) GetByKey(key string) (interface{}, error) {
 				return metafield.Value, nil
 			}
 		}
+
 	}
 
-	return nil, fmt.Errorf("could not find metafield with key %v", key)
```

Comment:
Opted for the empty string because if you don't find a metafield and it returns null you cannot cast it. I think this should not be so hard if it does not return a metafield. If you want to be certain the error exists then you can check for the error.